changeset:   3453:9e92aa37631a
tag:         tip
user:        Daniel Burrows <dburrows@debian.org>
date:        Thu Nov 19 15:33:11 2009 -0800
summary:     Initialize the temporary file module *after* logging so its startup messages don't cause a spurious error (Closes: #557150).

diff -r 31be843001d2 -r 9e92aa37631a src/main.cc
--- a/src/main.cc	Thu Nov 19 09:40:38 2009 -0800
+++ b/src/main.cc	Thu Nov 19 15:33:11 2009 -0800
@@ -540,8 +540,6 @@
 
   srandom(time(0));
 
-  temp::initialize("aptitude");
-
   using namespace log4cxx;
 
   // See earlier note
@@ -943,6 +941,8 @@
 							      log_file));
     }
 
+  temp::initialize("aptitude");
+
   const bool debug_search = aptcfg->FindB(PACKAGE "::CmdLine::Debug-Search", false);
 
   int curr_quiet = aptcfg->FindI("quiet", 0);

